src/main/java/org/akip/service/SubscriptionService.java

```diff
@@ -0,0 +1,151 @@
+package org.akip.service;
+
+
+import org.akip.domain.ProcessInstance;
+import org.akip.domain.ProcessInstanceSubscription;
+import org.akip.domain.enumeration.ProcessInstanceEventType;
+import org.akip.repository.ProcessInstanceRepository;
+import org.akip.repository.ProcessInstanceSubscriptionRepository;
+import org.akip.resolver.AkipUserDTO;
+import org.akip.resolver.UserResolver;
+import org.akip.service.mapper.ProcessInstanceMapper;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+@Service
+public class SubscriptionService {
```

Comment:
I'm still not convinced about this solution. Let's discuss together a better solution...